tests/test_s3.cc

```diff
@@ -0,0 +1,94 @@
+#include <experimental/filesystem>
+
+#include "gtest/gtest.h"
+
+#include "h5s3/s3.h"
+#include "process.h"
+
+namespace fs = std::experimental::filesystem;
+namespace s3 = h5s3::s3;
+
+using process = h5s3::testing::process;
+
+class S3Test : public ::testing::Test {
+public:
+    static std::string ACCESS_KEY;
+    static std::string SECRET_KEY;
+    static std::unique_ptr<process> minio_proc;
+
+protected:
+
+    static void SetUpTestCase() {
+        auto path = fs::temp_directory_path();
+        std::vector<std::string> args = {
+            "testbin/minio",
+            "server",
+            "--config-dir", "testbin/minio-config/",
+            "--address", "localhost:9999",
+            "/tmp/minio/",
+        };
+        process::environment env = {
+            {"MINIO_ACCESS_KEY", ACCESS_KEY},
+            {"MINIO_SECRET_KEY", SECRET_KEY},
+            {"PATH", std::getenv("PATH")},
+        };
+        minio_proc = std::make_unique<process>(args, env);
+        // HACK: Find a better way to determine that the minio server is up and
+        // running.
+        sleep(1);
+
+        std::vector<std::string> mc_args = {
+            "testbin/mc",
+            "--config-folder", "testbin/mc-config/", "--json",
+            "config",
+            "host",
+            "remove",
+            "h5s3-testing",
+        };
+        process::environment empty_env = {{"PATH", std::getenv("PATH")}};
```

Comment:
do we actually need the path? what is it looking for?

Reply:
minio shells out to something on startup.  without this I was getting errors from minio failing to find /bin/sh.